.github/workflows/infrastructure.yml

```diff
@@ -0,0 +1,173 @@
+# 2. Azure container registry - https://learn.microsoft.com/en-us/cli/azure/acr?view=azure-cli-latest#az-acr-create
+# 3. Azure container app environment - https://learn.microsoft.com/en-us/azure/container-apps/environment
+# 4. Azure container app - https://learn.microsoft.com/en-us/azure/container-apps/containers
+# 
+#
+# Execution
+# *********
+# GHA is only invoked when following conditions are satisfied:
+# 1. Push to the `infrastructure` branch only.
+# 2. Exact file path matches `.github/workflows/infrastructure.yml`.
+
+name: Infrastructure 🔨
+run-name: APIM base infrastructure build from ${{ github.repository }}
+
+on:
+  push:
+    branches: [ APIM-149-ia-c-dev-environment ]
+    paths: [ .github/workflows/infrastructure.yml ]
+
+env:
+  product: apim
+  environment: infrastructure
+  timezone: 'Europe/London'
+  # Deployment environment target i.e., `dev`, `staging`, `prod` 
+  target: ${{ vars.environment }}
+
+jobs:
+
+# 1. Setup infrastructure variables
+  setup:
+    name: Setup 🔧
+    runs-on: [ self-hosted, APIM, infrastructure ]
+    outputs:
+      environment: ${{ env.environment }}
+      timezone: ${{ env.timezone }}
+    steps:
+    - name: Environment 🧪
+      run: echo "Environment set to ${{ env.environment }}"
+
+    - name: Timezone 🌐
+      run: echo "Timezone set to ${{ env.timezone }}"
+
+# 2. Base infrastructure creation
+  base:
+    name: Base 🧱
+    needs: setup
+    environment: ${{ needs.setup.outputs.environment }}
+    runs-on: [ self-hosted, APIM, infrastructure ]
+    steps:
+
+    - name: Pre-production 💫
+      if: contains('["dev", "staging"]', env.target)
+      run: echo "type=Preproduction" >> $GITHUB_ENV
+
+    - name: Production 💫
+      if: ${{ 'prod' == env.target }}
+      run: echo "type=Production" >> $GITHUB_ENV
+
+    - name: Tags 🏷️
+      run: echo tags='Environment=${{ env.type }}' \
+        'Product=${{ env.product }}' \
+        'Region=${{ vars.REGION }}' \
+        'Team=development' >> $GITHUB_ENV
+
+    - name: Login 🔐
+      uses: azure/login@v1
+      with:
+        creds: ${{ secrets.AZURE_CREDENTIALS }}
+
+    - name: Azure defaults ✨
+      uses: azure/cli@v1.0.0
+      with:
+        inlineScript: |
+          # Basic
+          az configure --defaults location=${{ vars.REGION }}
+          az configure --defaults group=rg-${{ env.product }}-${{ env.target }}-001
+
+          # AZ CLI extensions upgrade
+          az extension add --name containerapp --upgrade
+
+    - name: Resource group 🏗️
+      uses: azure/cli@v1.0.0
+      with:
+        inlineScript: |
+          az group create \
+          --name rg-${{ env.product }}-${{ env.target }}-001 \
+          --tags ${{ env.tags }}
+
+    
+    - name: Container registry 📦️
+      uses: azure/cli@v1.0.0
+      with:
+        inlineScript: |
+          az acr create \
+          --name cr${{ env.product }}${{ env.target }}${{ vars.REGION }}001 \
```

Comment:
Same as the others, no need to add the egion

Reply:
Responded above regarding region.